### PR TITLE
Define class interfaces with tests

### PR DESCRIPTION
**Pull request by Edward-Boulderstone**

### CODE REVIEW

File: tests/test_wallet.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+from milestone_2.Wallet import Wallet
+
+
+@pytest.fixture
+def new_wallet() -> Wallet:
+    return Wallet()
+
+
+def empty_wallet() -> Wallet:
```
Comment (vbeardow) [inline tests/test_wallet.py:12]: Does this need a @fixture?

File: tests/test_hand.py [hunk 2]
```diff
@@ -27,15 +27,20 @@ def six() -> TestCard:
...
+def sixteen_test_hand(six: TestCard, ten: TestCard) -> Hand:
     return Hand(six, ten)
 
 
+def test_hand(*card_values: int) -> Hand:
+    TestCards = [TestCard(card_value) for card_value in card_values]
+    return Hand(*TestCards)
+
+
 def test_starting_hand_value(ace_hand: Hand, sixteen_hand: Hand) -> None:
```
Comment (lewiswilkins) [inline tests/test_hand.py]: Quite a lot of different assertions going on in one test. Really, tests should be testing only one thing at a time.

File: tests/test_wallet.py (hunk 1 above)
Comment (Edward-Boulderstone) [reply tests/test_wallet.py:12]: Yes, thank you

File: tests/test_hand.py [hunk 3]
```diff
@@ -55,3 +60,69 @@ def test_is_bust(
     assert not sixteen_hand.is_bust()
     assert not ace_hand.draw(ten).is_bust()
     assert sixteen_hand.draw(six).is_bust()
+
+
+def test_comparison_of_standard_hands() -> None:
+    hand_15 = test_hand(6, 9)
```
Comment (vbeardow) [inline tests/test_hand.py:88]: Not sure I understand what you are testing here.

Your `test_starting_hand_value` should be enough to verify the total value of the hand. If it verifies that `test_hand(6, 9)` and `test_hand(2, 3)` gives the correct values 15 and 5, your assert statement on line 70 only asserts that 15 > 5 which will always be true.

**Review by vbeardow — COMMENTED** [review]
Not sure if I understand the purpose of some of these tests. It seems like they are just comparing two values. I think it should be sufficient to test that the value they are returning is correct, otherwise the tests are simply asserting 15 > 5 for example which is always true.

File: tests/test_hand.py (hunk 3 above)
Comment (Edward-Boulderstone) [reply tests/test_hand.py:88]: I want the hand class to be able to have it's own internal comparator. So for standard non-ace, non-bust hands, I want to verify it just compares their values directly.

Comment (Edward-Boulderstone) [reply tests/test_hand.py:88]: I don't know if that's overkill or not.

File: tests/test_hand.py (hunk 2 above)
Comment (Edward-Boulderstone) [reply tests/test_hand.py]: Thank you, I have split the test_starting_hand_value into two seperate test cases

File: tests/test_hand.py (hunk 3 above)
Comment (vbeardow) [reply tests/test_hand.py:88]: In that case I think the test should then compare that the function doing the comparison outputs/executes the correct thing. Something like below maybe ? Correct me if I've misunderstood

```
def hand_compare(hand1, hand2):
    if hand1 > hand2:
        return True
    else:
        return False
    
def test_compare():
    assert hand_compare(3, 4) == False
   ```

Comment (Edward-Boulderstone) [reply tests/test_hand.py:88]: Is that not what I am doing? I was going write the hand_compare method using the __eq__, __gr__, __lt__ inbuilt methods

Comment (vbeardow) [reply tests/test_hand.py:88]: OK i understand now, you are comparing the hands themselves and not the value of the hands so it makes sense.

**Pull request merged**